definitions.h

```diff
@@ -240,8 +243,42 @@ class Attributes
             // Your code goes here when clipping is implemented
         }
 
-        PIXEL color;
-};	
+        Attributes &operator *= (const double rhs[][4]);// throw (const char *);
+        Attributes &operator *= (const double rhs[]);// throw (const char *);
+};
+
+Attributes &Attributes::operator *= (const double rhs[][4]) //throw (const char *)
```

Comment:
I am not sure, but I don't think we need to overload the operators for Attributes.  Perhaps what you need to do is overload the operators for your Matrix Transform class.  I have not yet looked at your Matrix Transform class.